app/src/main/java/com/hattrick/myapplication/presentation/fragment/SampleFragment.kt

```diff
@@ -8,4 +12,14 @@ class SampleFragment : BaseFragment<SampleViewModel, FragmentSampleBinding>(Samp
 
     override fun getViewBinding() = FragmentSampleBinding.inflate(layoutInflater)
 
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        binding.navButton.setOnClickListener {
```

Comment:
Let’s add a setupNavigation() method on base fragment, override it and put everything concerning navigation on that method.